src/python/WMCore/Services/DDM/DDM.py

```diff
@@ -0,0 +1,286 @@
+        callname = "DDM-{}".format(api)
+
+        result = None
+        try:
+            # TODO: here/getResult to send the request parameters to the logger
+            result = self._getResult(ddmReq, apiUrl, callname)
+        except Exception as e:
+            msg = "ERROR: sending data to DDM: \n"
+            msg += "API: {}\n".format(ddmReq.api)
+            msg += "ddmReq: {}\n".format(pformat(ddmReq))
+            msg += "ERROR: {}".format(str(e))
+            self['logger'].debug(msg)
+            return None
+        return result
+
+    def makeAggRequests(self, ddmReqList, aggKey=None):
+        """
+        _makeAggRequest_
+
+        Makes several Aggregated requests to DDM. Takes the list of objects,
+        instances of ddmReqTemplate and aggregates it based on a selected key.
+        It iterates through the list and compares all to all. Uses the
+        ddmReqTemplate internal Equivalence method for comparison with the
+        aggregation key excluded. If a match is found the two requests are
+        squeezed into one. The final result should be a new list of unique
+        ddmRequests each one of them should contain all similar ddmRequests
+        from the initial list with the values of the aggregation key accumulated
+        in a list under the same key name. Creates a request per ddmReqTemplate
+        from the list with aggregated templates.
+        NOTE:
+            In this method the api is always fetched from the template.
+        TODO:
+            So far it is working basically for the 'copy' API and due to that
+            the aggregation key is always a dictionary so no explicit check is
+            done for the type. In case we decide to use it we should develop
+            this to the end with the proper type checks.
+
+        :ddmReqList: List of objects, instances of DDMReqTemplate class.
+        :aggKey:    String representing the Aggregation key.
+        :return:     List of Dictionaries containing the results from DDM
+
+        """
+        if not isinstance(ddmReqList, list) or len(ddmReqList) == 0:
+            return None
+
+        ddmReqListAgg = []
+
+        # populate the first element in the aggregated list
+        ddmReqListAgg.append(ddmReqList.pop())
+
+        # feed the rest
+        while len(ddmReqList) != 0:
+            ddmReq = ddmReqList.pop()
+            for aggReq in ddmReqListAgg:
+                if ddmReq.isEqual(aggReq, aggKey):
+                    # Check if the two objects are not references to one and the
+                    # same object. Only then copy the values of the aggKey,
+                    # otherwise we will enter an endless cycle.
+                    if ddmReq is not aggReq and aggKey is not None:
+                        for i in ddmReq[aggKey]:
+                            aggReq[aggKey].append(i)
+                    del(ddmReq)
+                    break
+            if 'ddmReq' in locals():
```

Comment:
This check is too nasty!

Reply:
We must check somehow that the object was not already matched, aggregated and dropped from the `for` cycle above. And if there was not a match found for it, this means it still exists here as a new and unique object and needs to be added as such at the end of the aggregated list, so that the next cycle it will be included for checks with the rest of the objects. If you have a better idea how to do this check I am open for suggestions.

Reply:
I'm not sure how "pythonic" it is to use the locals function in the code (besides for debugging purposes)... So, I would actually use a flag to signalize whether we need to append the object or not.

Reply:
It will add one more line  and achieving the same result, but will make the code much clear and obvious while reading. I agree. WiIl change that too.

Reply:
Done